src/app/tests/TestAttributePersistenceProvider.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements unit tests for AttributePersistenceProvider
+ *
+ */
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/support/TestPersistentStorageDelegate.h>
+#include <app/DefaultAttributePersistenceProvider.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::TLV;
+
+const ConcreteAttributePath TestConcretePath = ConcreteAttributePath(1, 1, 1);
+
+namespace {
+
+/**
+ *  Set up the test suite.
+ */
+int Test_Setup(void * inContext)
+{
+    CHIP_ERROR error = chip::Platform::MemoryInit();
+    VerifyOrReturnError(error == CHIP_NO_ERROR, FAILURE);
+    return SUCCESS;
+}
+
+/**
+ *  Tear down the test suite.
+ */
+int Test_Teardown(void * inContext)
+{
+    chip::Platform::MemoryShutdown();
+    return SUCCESS;
+}
+
+/**
+ * Tests the storage and retrival of data from the KVS as ByteSpan
+ */
+void TestStorageAndRetrivalByteSpans(nlTestSuite * inSuite, void * inContext)
+{
+    TestPersistentStorageDelegate storageDelegate;
+    DefaultAttributePersistenceProvider persistenceProvider;
+
+    // todo Failure before Init
+
+    // Init
+    CHIP_ERROR err = persistenceProvider.Init(&storageDelegate);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    // Store ByteSpan of size 1
+    uint8_t setArray[1] = {0x42};
+    ByteSpan setValue(setArray, sizeof(setArray));
```

Comment:
You can replace all of these with `ByteSpan setValue{setArray};` --> there is a templated constructor for direct-from-array.

Also suggest renaming to avoid "set/get" prefixes which are usually expected of setters/getters rather than particular values. e.g. `setValue --> value`, `getValue --> valueReadBack`

Reply:
Done.